### PR TITLE
libsamplerate 0.2.2

### PR DESCRIPTION
**Pull request by guillaumekh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- Upgraded libsamplerate formula to 0.2.2
- Replaced livecheck & homepage since the [official website's Download page](http://www.mega-nerd.com/SRC/download.html) is now lagging behind GitHub releases
- Moved to cmake
- Replaced `sndfile-resample` w/ `simple_test` since it [moved to another repo](https://github.com/libsndfile/libsamplerate/blob/master/NEWS#L26)
- Removed libsndfile dependency since we don't rely on `sndfile-resample` anymore
- Removed no long required patch

### CODE REVIEW

**Comment by guillaumekh** [comment]
I don't understand yet why dylibs are not found when running the test from bottle 🤔

**Comment by SMillerDev** [comment]
> I don't understand yet why dylibs are not found when running the test from bottle 🤔. Is there something I'm missing?

Because it's in a completely different place there then when you install it. You should make your own minimal test in the test block instead of copying a binary.

**Comment by guillaumekh** [comment]
>Because it's in a completely different place there then when you install it.

Ah got it. I did not understand how shared libraries are located in MacOS. `man dyld` just shed some light but I need to read more.

>You should make your own minimal test in the test block instead of copying a binary.

OK. For what it's worth, this is pretty much what the formula was doing beforehand with `sndfile-resample`.

I've avoided writing a small C test program myself because I'm really inexperienced with C. I might be able to extract a minimally viable test from the `simple_test` [source](https://github.com/libsndfile/libsamplerate/blob/master/tests/simple_test.c) though. I'll give it a try.

**Comment by guillaumekh** [comment]
I've rewritten the test.
The fast arm64 CI jobs passed but others seem to have stalled during the `test-bot --only-formulae-dependents` step & timed out. I'm not sure why.
Please let me know if further work is required.

**Comment by guillaumekh** [comment]
All arm64 CI jobs pass except the x86-64 jobs which time out during the `test-bot --only-formulae-dependents` step because of the high number and size of packages relying on this library (e.g. ffmpeg).

Is labelling this PR as "long build" warranted?

**Comment by chenrui333** [comment]
yes! long build needed.

File: Formula/libsamplerate.rb [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+  sha256 "16e881487f184250deb4fcb60432d7556ab12cb58caea71ef23960aec6c0405a"
   license "BSD-2-Clause"
-  revision 1
 
   livecheck do
-    url "http://www.mega-nerd.com/SRC/download.html"
-    regex(/href=.*?libsamplerate[._-]v?(\d+(?:\.\d+)+)\.t[^.]+(?:\.[^.]+)?["' >]/i)
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
   end
```
Comment (chenrui333) [inline Formula/libsamplerate.rb:11]: we can just remove the livecheck to use the default one.

Comment (chenrui333) [reply Formula/libsamplerate.rb:11]: I just tested the removal, working fine

```
$ brew livecheck libsamplerate
libsamplerate: 0.2.2 ==> 0.2.2
```

**Comment by chenrui333** [comment]
LGTM, will merge after icu4c PR

- jack test failure on 12-arm
- audacious and bazarr build failure on 11 due to github networking issue

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@guillaumekh, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**